src/vs/gitpod/browser/gitpodInsightsAppender.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable code-import-patterns */
+/* eslint-disable header/header */
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { RemoteTrackMessage } from '@gitpod/gitpod-protocol/lib/analytics';
+import { ITelemetryAppender } from 'vs/platform/telemetry/common/telemetryUtils';
+
+export class GitpodInsightsAppender implements ITelemetryAppender {
+	constructor() {
+
+	}
+	public log(eventName: string, data: any): void {
+		const trackMessage = mapTelemetryData(eventName, data);
+		if (!trackMessage) {
+			return;
+		}
+		window.postMessage({ type: 'vscode_telemetry', event: trackMessage.event, properties: trackMessage.properties }, '*');
```

Comment:
nice idea

I was thinking about using production untrusted, but this seems to be better